syntax = "proto3";

message Empty {
}

service UserAPI {
    rpc StartJob(StartJobRequest) returns (StartJobReply) {}
    rpc LastJobStatus(Empty) returns (LastJobStatusReply) {}
}

message StartJobRequest {
    string inputLocation = 1;
}

message StartJobReply {
    string jobUuid = 1;
}

message JobStatusRequest {
    string inputLocation = 1;
}

message LastJobStatusReply {
    message JobStatus {
        string jobUuid = 1;
        bool finished = 2;
        // current step? more detailed node status?
    }
    JobStatus status = 1;
}

service NodeAPI {
    rpc StartStep(StartStepRequest) returns (StartStepReply) {}
    rpc NodeStatus(Empty) returns (NodeStatusReply) {}
}

message StartStepRequest {
    string inputLocation = 1;
    string outputLocation = 2;
    string stepId = 3;
}

message StartStepReply {
    bool ok = 1;
}

message NodeStatusReply {
    enum NodeStatusEnum {
        Unspecified = 0;
        Ok = 1;
        Working = 2;
        Failure = 3;
    }
    NodeStatusEnum status = 1;
    string errorMessage = 2;
}
